src/server.js

```diff
@@ -0,0 +1,59 @@
+require('require-self-ref')
+const {checkWatcher, startNPMWatcher} = require('./watchNPM')
+const redis = require('redis')
+const logger = require('~/src/logging').logger(module)
+const config = require('~/src/config')
+const queue = require('windbreaker-service-util/queue')
+const Promise = require('bluebird')
+config.load()
+
+;(async () => {
+  let connection = null
+  let channel = null
+  let client = null
+  const AmqUrl = config.getAmqUrl()
+
+  const setup = async function () {
+    try {
+      connection = await queue.createConnection({
+        logger,
+        AmqUrl
+      })
+      channel = await connection.createChannel()
+      await channel.assertQueue(config.getQueueName())
+      client = redis.createClient(config.getRedisURL())
+      logger.info('setup successful')
+    } catch (error) {
+      logger.error(error)
+      logger.error('cleaning up and restarting setup')
+      client = null
+      if (channel) channel.close()
+      channel = null
+      if (connection) connection.close()
+      connection = null
+      await Promise.delay(1000).then(async function () {
+        await setup()
+      })
+    }
+  }
```

Comment:
The `queue` object above exposes a `createProducer` function that you should be using. See [this](https://github.com/windbreaker-io/windbreaker-service-util/blob/master/queue/QueueProducer.js) for the definition of the producer you will be using and check out some of the integration tests for an idea on how you can use it. You will have to listen for errors to know when the producer has lost a connection and needs to be reinitialized.